### PR TITLE
#36 Profile Section inital: created profileapp, for service-seeker..

### PR DESCRIPTION
**Pull request by deepjyotk**
… profile and service service provider profile

### CODE REVIEW

File: src/profileapp/models/service_seeker_model.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+class ServiceSeeker(models.Model):
+    user = models.OneToOneField(CustomUser, on_delete=models.CASCADE)
+    bookmarked_services = models.JSONField(
+        default=list, blank=True, null=True
+    )  # Allow null values
+    location_preference = models.CharField(
+        max_length=255, help_text="Preferred borough or location in NYC"
+    )
```
Comment (ahhcash) [inline src/profileapp/models/service_seeker_model.py:12]: So this model will be present in our sqlite3 db. Is that correct?

File: src/profileapp/__init__.py
Comment (ahhcash) [inline src/profileapp/__init__.py:1]: Should we integrate this into the accounts app itself?

File: src/accounts/views.py [hunk 2]
```diff
@@ -16,6 +18,7 @@ def register(request):
             if user.user_type == "service_provider":
                 return redirect("service_provider_dashboard")
             else:
+                ServiceSeeker.objects.create(user=user)
```
Comment (ahhcash) [inline src/accounts/views.py:21]: I think the ServiceSeeker and user(user_type=="seeker") might introduce redundancy

File: src/profileapp/forms/service_provider_form.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+
+class ServiceProviderForm(forms.Form):
+    business_name = forms.CharField(max_length=255)
+    business_address = forms.CharField(max_length=255)
+    service_description = forms.CharField(widget=forms.Textarea)
+    service_categories = forms.CharField(max_length=255)
+    service_location = forms.CharField(max_length=255)
+    availability = forms.CharField(max_length=255)
+    price_range = forms.CharField(max_length=100)
```
Comment (ahhcash) [inline src/profileapp/forms/service_provider_form.py:11]: nice! django forms seem to be pretty neat!

**Review by ahhcash — CHANGES_REQUESTED** [review]
My main concern is the separate app for profiles

File: src/profileapp/__init__.py
Comment (deepjyotk) [reply src/profileapp/__init__.py:1]: According to me, the accounts app should handle authentication and related models. We can reference those models in other apps (as demonstrated in the ServiceSeeker model, where the user field references the CustomUser model from the accounts app).

**Comment by deepjyotk** [comment]
Ideally, the profiles for service seekers and service providers should be distinct. However, I don't see an issue with both profiles being managed within the same app. Since we're structuring the project by apps based on epics (features), and profiles are a feature, implementing them together for both user types will save time. This way, we won't need to create separate user stories for each profile type.

Additionally, this is just the initial code, and the feature isn't fully implemented yet.

**Comment by ahhcash** [comment]
> Since we're structuring the project by apps based on epics (features), and profiles are a feature, implementing them together for both user types will save time. 

Yes, makes sense. I think we're on the same page here. Refactoring `profileapp` into the existing `accounts` app, keeping the additional models you created (like `ServiceSeekerModel`), should help with reducing complexity.

>Additionally, this is just the initial code, and the feature isn't fully implemented yet.

Sure, we can iterate and revise as the feature evolves

File: src/profileapp/__init__.py
Comment (ahhcash) [reply src/profileapp/__init__.py:1]: I believe we had a brief discussion earlier about trying our best to maintain one app per epic. This story is part of #8 , and iirc, the motivation was that the user authentication and profile info are related stories. That's why we put them in the same epic.

File: src/accounts/views.py (hunk 2 above)
Comment (ahhcash) [reply src/accounts/views.py:21]: if a refactor is done to merge the profileapp into the accounts app (and maybe rename it to user_mgmt or something along those lines), keeping a separate ServiceSeeker model makes sense. will resolve this comment based on further review

**Pull request closed without merging**